app/controllers/agents/users_controller.rb

```diff
@@ -135,4 +152,11 @@ def search_params
   def set_user
     @user = policy_scope(User).find(params[:id])
   end
+
+  def set_user_profile(user, organisation)
```

Comment:
vraiment pas fan du tout de cette methode compliquée meme si je comprends d'ou vient l'idee apres la discussion qu'on avait eu. 

de plus cette maniere genere des records fantomes dans la db, ca cree en fait deux user_profiles dont un avec un user_id vide. 

<img width="425" alt="Screenshot 2020-06-15 at 17 04 10" src="https://user-images.githubusercontent.com/883348/84677856-979a0a80-af2f-11ea-9650-719b611df3d2.png">

il faudrait d'ailleurs rajouter une validation pour empecher ca dans le modele UserProfile

je te propose une autre maniere de faire dans ce patch : 
<details>
<summary>user_profiles.patch</summary>

```
diff --git a/app/controllers/agents/users_controller.rb b/app/controllers/agents/users_controller.rb
index 88c52ae3..298147ca 100644
--- a/app/controllers/agents/users_controller.rb
+++ b/app/controllers/agents/users_controller.rb
@@ -19,7 +19,7 @@ class Agents::UsersController < AgentAuthController
 
   def new
     @user = User.new
-    @user.organisation_ids = [current_organisation.id]
+    @user.user_profiles.build(organisation: current_organisation)
     if params[:responsible_id].present?
       @user.responsible = policy_scope(User).find(params[:responsible_id])
     end
@@ -111,11 +111,8 @@ class Agents::UsersController < AgentAuthController
 
   def prepare_create
     @user = User.new(user_params)
-    authorize(@user.responsible) if @user.responsible.present?
-
-    @user.user_profiles = [set_user_profile(@user, current_organisation)]
-
     @user.invited_by = current_agent
+    authorize(@user.responsible) if @user.responsible.present?
     @organisation = current_organisation
   end
 
@@ -141,7 +138,8 @@ class Agents::UsersController < AgentAuthController
       :number_of_children,
       :invite_on_create,
       :responsible_id,
-      user_profiles_attributes: [:notes, :logement, :id], agent_ids: []
+      user_profiles_attributes: [:notes, :logement, :id, :organisation_id],
+      agent_ids: []
     )
   end
 
@@ -152,11 +150,4 @@ class Agents::UsersController < AgentAuthController
   def set_user
     @user = policy_scope(User).find(params[:id])
   end
-
-  def set_user_profile(user, organisation)
-    user.organisation_ids = [organisation.id]
-    profile = user.user_profiles.first
-    profile.organisation = organisation
-    profile
-  end
 end
diff --git a/app/policies/agent/user_policy.rb b/app/policies/agent/user_policy.rb
index 57aa2ed4..0611fc42 100644
--- a/app/policies/agent/user_policy.rb
+++ b/app/policies/agent/user_policy.rb
@@ -24,4 +24,19 @@ class Agent::UserPolicy < DefaultAgentPolicy
       scope.joins(:organisations).where(organisations: { id: @context.organisation.id })
     end
   end
+
+  protected
+
+  def same_org?
+    # we cannot use @record.organisation_ids for Users because it uses a
+    # has_many through: :user_profiles, and `collection_singular_ids` only
+    # returns ids for persisted join records so it doesn't work for new records
+    # nor updates. we cannot either use pluck for the same reason
+    @record.user_profiles.map(&:organisation_id).include?(@context.organisation.id)
+
+    # also, this is not strictly speaking correct. this only checks that the
+    # resulting user will belong to the current organisation, but it should also
+    # check that you're not making any updates (add or remove) for other
+    # organisations.
+  end
 end
diff --git a/app/views/agents/users/_form.html.slim b/app/views/agents/users/_form.html.slim
index a8c8c063..061cf8fe 100644
--- a/app/views/agents/users/_form.html.slim
+++ b/app/views/agents/users/_form.html.slim
@@ -37,6 +37,7 @@
       .col-md-6= f.input :number_of_children, input_html: { min: '0', max: '15', step: 'any' }
 
     = f.simple_fields_for :user_profiles, user_profile  do |n|
+      = n.input :organisation_id, as: :hidden
       = n.input :logement, collection: UserProfile.human_enum_collection(:logement)
       = n.input :notes, input_html: { rows: 6 }

```
</details>

ca me parait plus propre et ca ne genere pas de records fantomes